Change BEAM lights and sights to accept width instead of arc

kwvanderlinde opened the pull request:
### Identify the Bug or Feature request

Improves #4448

### Description of the Change

Instead of an `arc=` parameter, BEAM lights and sights now take a `width=` parameter measured in map units (same as ranges).

Also includes clean up for `SightType` and fixed sight output in `getInfo("campaign")`.

### Possible Drawbacks

It's possible to make beams so wide they don't look like beams anymore.

### Documentation Notes

BEAM lights can take `width=` parameter to set the thickness of the beam.

### Release Notes

N/A

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/RPTools/maptool/4516)
<!-- Reviewable:end -->

kwvanderlinde committed:
Change BEAM lights and sights to accept width instead of arc

Width is measure in map units, so works the same as ranges.

kwvanderlinde committed:
Clean up SightType usage

- Remove unused methods and constructors.
- Remove all mutators (the few in use were only at construction time anyways).
- Move the constructor to be above the methods, not in the middle.
- Move the personal light source to be the last constructor parameter, for better literals.
- `SightType.shape` is no longer allowed to be `null`.
- Remove default constructor, use the real constructor in `fromDto()`
- Initial sight type improvements:
  - They are written as `SightType`, not `Object[]`.
  - Darkvision no longer coupled to the Generic light sources

kwvanderlinde committed:
Fix up sights in getInfo("campaign")

bubblobill commented:
The code looks good to me, but I couldn't find a default value for width in the parsing (I just got out of bed so may have missed it). A default width should be thin enough to be useful but thick enough to be visible.
Something like this, fiddlethe value to produce a satisfactory result.
```
this.width = (width == 0) ? 0.2 : width;
this.arcAngle = (arcAngle == 0) ? 90 : arcAngle;
```
Aim for something like this
![image](https://github.com/RPTools/maptool/assets/45483160/4bb7c0ec-b48e-4f7e-94b6-7f402f998259)

kwvanderlinde committed:
Add default behaviour for zero-width beams

When getting its shaped area, the width will be forced to be at least one pixel wide.

kwvanderlinde commented:
@bubblobill Any default value or lower bound for width won't necessarily be meaningful for all maps as they can have wildly settings. So instead I changed the meaning of width=0 (or any tiny width) to mean a rectangle that's just about as thin as possible, something we can determine dynamically.

![image](https://github.com/RPTools/maptool/assets/7492219/8e508e07-8b0d-44ab-87ba-2b9fc3f6c6a5)
